### PR TITLE
De-duplicate cmake

### PR DESCRIPTION
**Pull request by gbiggs**
This PR moves duplicated CMake code for the packages `depthai_ros_msgs` and `depthai_ros_driver` into CMake macros. The `CMakeLists.txt` files are significantly reduced in complexity, and the macros will be reusable in future nodes.

### CODE REVIEW

File: depthai_ros_driver/CMakeLists.txt [hunk 1]
```diff
@@ -1,171 +1,81 @@
...
+# paths used to find ROS packages, so we have to search for them first and then
+# add the external project
+find_dependent_packages(${PKG_DEPS})
+
+add_subdirectory(${DEPTHAI_CORE} depthai-core)
+
+add_ros_node(depthai_ros_driver
+  SKIP_PACKAGE_SEARCH
+  PROJECT_NAME ${PROJECT_NAME}
+  COMPONENT_CLASS "rr::DepthAIBaseRos2"
```
Comment (youliangtan) [inline depthai_ros_driver/CMakeLists.txt]: Is it better if we name it as `ROS2_COMPONENT_CLASS` and place it below `ROS1_COMPONENT_SOURCES` for readability

File: depthai_ros_driver/CMakeLists.txt [hunk 2]
```diff
@@ -1,171 +1,81 @@
...
-
-    install(TARGETS
-        ${PROJECT_NAME}_node
-        DESTINATION lib/${PROJECT_NAME})
-
-    ament_package()
-else()
-    message(VERBOSE "ROS version is $ENV{ROS_VERSION}, skipping this ROS2 ${PROJECT_NAME}")
+if($ENV{ROS_VERSION} STREQUAL 1)
+  install(FILES nodelet_plugins.xml DESTINATION ${depthai_ros_driver_PACKAGE_SHARE_DESTINATION})
```
Comment (youliangtan) [inline depthai_ros_driver/CMakeLists.txt:80]: Is it possible to move this into the macro? (let `DIRECTORIES` accepts files??) I guess there's no clean way to do it too.

Comment (gbiggs) [reply depthai_ros_driver/CMakeLists.txt:80]: This file is ROS 1-specific, whereas the `DIRECTORIES` argument is accepting directories to install in both ROS 1 and ROS 2.

File: depthai_ros_driver/CMakeLists.txt (hunk 1 above)
Comment (gbiggs) [reply depthai_ros_driver/CMakeLists.txt]: I've renamed the parameter. The ordering doesn't matter as it's a keyword parameter.

**Review by youliangtan — APPROVED** [review]
Tested, LGTM!

File: depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    add_library(${name}_lib SHARED ${_ARG_SOURCES} ${_ARG_ROS2_SOURCES})
+    foreach(sys_dep ${_ARG_SYSTEM_DEPS})
+      target_link_libraries(${name}_lib PUBLIC ${${sys_dep}_LIBRARIES})
+    endforeach()
+    foreach(target_dep ${_ARG_TARGET_DEPS})
+      target_link_libraries(${name}_lib PUBLIC ${target_dep})
+    endforeach()
+    ament_target_dependencies(${name}_lib PUBLIC ${_ARG_PKG_DEPS})
+
+    rclcpp_components_register_nodes(${name}_lib ${_ARG_COMPONENT_CLASS})
```
Comment (youliangtan) [inline depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake]: should be `${_ARG_ROS2_COMPONENT_CLASS}`

Comment (gbiggs) [reply depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake]: Fixed in 5d03b43

File: depthai_ros_driver/CMakeLists.txt [hunk 4]
```diff
@@ -1,171 +1,81 @@
...
+find_package(Boost REQUIRED COMPONENTS system)
 
-set (PKG_DEPS_COMMON
-    camera_info_manager
-    cv_bridge
-    depthai_ros_msgs
-    image_transport
-    sensor_msgs
-    std_msgs
+set(PKG_DEPS
```
Comment (kunaltyagi) [inline depthai_ros_driver/CMakeLists.txt]:
```suggestion
set(COMMON_PKG_DEPS
```

**Review by kunaltyagi — APPROVED** [review]
Rest looks good to me :)

File: depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake (hunk 3 above)
Comment (gbiggs) [reply depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake]: Fixed in 4a36aa0

File: depthai_ros_driver/CMakeLists.txt (hunk 4 above)
Comment (gbiggs) [reply depthai_ros_driver/CMakeLists.txt]: Fixed in 4a36aa0

**Pull request merged**